src/System.Net.Security/src/System/Net/SecurityStatusAdapterPal.Windows.cs

```diff
@@ -0,0 +1,107 @@
+            new SecurityStatusMapping(Interop.SecurityStatus.OutOfMemory, SecurityStatusPal.OutOfMemory),
+            new SecurityStatusMapping(Interop.SecurityStatus.OutOfSequence, SecurityStatusPal.OutOfSequence),
+            new SecurityStatusMapping(Interop.SecurityStatus.PackageNotFound, SecurityStatusPal.PackageNotFound),
+            new SecurityStatusMapping(Interop.SecurityStatus.MessageAltered, SecurityStatusPal.MessageAltered),
+            new SecurityStatusMapping(Interop.SecurityStatus.NoCredentials, SecurityStatusPal.NoCredentials),
+            new SecurityStatusMapping(Interop.SecurityStatus.QopNotSupported, SecurityStatusPal.QopNotSupported),
+            new SecurityStatusMapping(Interop.SecurityStatus.Renegotiate, SecurityStatusPal.Renegotiate),
+            new SecurityStatusMapping(Interop.SecurityStatus.SecurityQosFailed, SecurityStatusPal.SecurityQosFailed),
+            new SecurityStatusMapping(Interop.SecurityStatus.SmartcardLogonRequired, SecurityStatusPal.SmartcardLogonRequired),
+            new SecurityStatusMapping(Interop.SecurityStatus.TargetUnknown, SecurityStatusPal.TargetUnknown),
+            new SecurityStatusMapping(Interop.SecurityStatus.TimeSkew, SecurityStatusPal.TimeSkew),
+            new SecurityStatusMapping(Interop.SecurityStatus.UnknownCredentials, SecurityStatusPal.UnknownCredentials),
+            new SecurityStatusMapping(Interop.SecurityStatus.UnsupportedPreauth, SecurityStatusPal.UnsupportedPreauth),
+            new SecurityStatusMapping(Interop.SecurityStatus.Unsupported, SecurityStatusPal.Unsupported),
+            new SecurityStatusMapping(Interop.SecurityStatus.UntrustedRoot, SecurityStatusPal.UntrustedRoot),
+            new SecurityStatusMapping(Interop.SecurityStatus.WrongPrincipal, SecurityStatusPal.WrongPrincipal),
+        };
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromWin32Int(int win32SecurityStatus)
+        {
+            return GetSecurityStatusPalFromInterop((Interop.SecurityStatus) win32SecurityStatus);
+        }
+
+        internal static SecurityStatusPal GetSecurityStatusPalFromInterop(Interop.SecurityStatus win32SecurityStatus)
+        {
+            foreach (SecurityStatusMapping mapping in s_securityStatusMapping)
+            {
+                if (win32SecurityStatus == mapping.Win32Status)
+                {
+                    return mapping.SecurityStatus;
+                }
+            }
+
+            //no match found for SecurityStatus value
+            Debug.Fail("Unknown Interop.SecurityStatus value: " + win32SecurityStatus);
+            throw new InternalException();
+        }
+
+        internal static Interop.SecurityStatus GetInteropFromSecurityStatusPal(SecurityStatusPal status)
+        {
+            if (status == SecurityStatusPal.NotSet)
+            {
+                Debug.Fail("SecurityStatus NotSet");
+                throw new InternalException();
+            }
```

Comment:
Is this check necessary?  If status is NotSet, won't we just fail to find a mapping and end up in the same Debug.Fail and throw below?